modules/core/src/main/java/org/apache/ignite/internal/binary/BinaryUtils.java

```diff
@@ -2554,6 +2591,41 @@ public static void validateEnumValues(String typeName, @Nullable Map<String, Int
         return mergedMap;
     }
 
+    /**
+     * @param obj {@link BinaryArray} or {@code Object[]}.
+     * @return Objects array.
+     */
+    public static Object[] rawArrayFromBinary(Object obj) {
+        if (obj instanceof BinaryArray)
+            // We want raw data(no deserialization).
+            return ((BinaryArray)obj).array();
+        else
+            // This can happen even in BinaryArray.USE_TYPED_ARRAY = true.
+            // In case user pass special array type to arguments, String[], for example.
+            return (Object[])obj;
+    }
+
+    /** */
+    public static Object[] rawArrayInArgs(Object[] args, boolean keepBinary) {
+        if (args == null)
+            return args;
+
+        for (int i = 0; i < args.length; i++) {
+            if (args[i] instanceof BinaryArray) {
+                BinaryArray arr = (BinaryArray)args[i];
+
+                args[i] = keepBinary ? arr.array() : arr.deserialize();
```

Comment:
Why only `BinaryArray`'s are deserialized? What about regular binary objects? With such a behavior binary objects and regular objects will coexist in the same args array, I think it's error-prone.

Reply:
Agree. Removed